include/libcyphal/executor.hpp

```diff
@@ -29,164 +29,49 @@ using IExecutorTypeIdType = cetl::
 class IExecutor : public cetl::rtti_helper<IExecutorTypeIdType>
 {
 public:
-    /// @brief Defines a callback abstraction umbrella type.
+    /// NOSONAR cpp:S4963 - we do directly handle callback resource here.
     ///
-    struct Callback
+    class Callback
     {
-        /// @brief Defines low level unique identifier for a callback.
-        ///
-        /// 64 bits should be enough to ensure uniqueness of the callback identifier
-        /// by applying some simple rules (e.g. incrementing counter).
-        ///
-        using Id = std::uint64_t;
-
+    public:
         /// @brief Defines possible variants of callback schedules.
         ///
         struct Schedule
         {
+            /// @brief Defines schedule which will NOT execute callback function.
+            ///
+            /// Useful as a default value for a non-engaged callback schedule.
+            /// Also can be used to cancel previously scheduled callback.
+            ///
+            struct None
+            {};
```

Comment:
This special case doesn't look convincing. I understand that there is some value in being able to unschedule a callback but as far as I understand by now we haven't identified an explicit need for that, and it would be easy to retrofit -- if needed -- later. The troubling thing here is that an extension of any well-structured set of entities with a special case often tends to break the design in unexpected ways at some point.

>*But I couldn’t resist the temptation to put in a null reference, simply because it was so easy to implement.*

Reply:
In the next pr I'll refactor this in favor of optional schedule.

Reply:
https://github.com/OpenCyphal-Garage/libcyphal/issues/372